### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by JohnTitor**
* Use `match` instead of `if` chain
* Remove redundant `into_iter()`
* Use `copied()` instead of `map()`

etc.

### CODE REVIEW

File: src/librustc_apfloat/ieee.rs [hunk 1]
```diff
@@ -388,52 +388,54 @@ impl<S: Semantics> fmt::Display for IeeeFloat<S> {
         let _: Loss = sig::shift_right(&mut sig, &mut exp, trailing_zeros as usize);
```
Comment (Centril) [inline src/librustc_apfloat/ieee.rs]: cc @eddyb

Comment (eddyb) [reply src/librustc_apfloat/ieee.rs]: Thanks for the ping!

@JohnTitor The indentation change is unfortunate, and I'm not sure using `Ordering` makes this more readable. But also, this code was ported from C++, and I'd rather not diverge too much (main exception being abstractions that avoid code duplication etc.).

File: src/librustc_apfloat/ieee.rs [hunk 2]
```diff
@@ -1199,8 +1201,8 @@ impl<S: Semantics> Float for IeeeFloat<S> {
         }
 
         // Handle a leading minus sign.
-        let minus = s.starts_with("-");
-        if minus || s.starts_with("+") {
+        let minus = s.starts_with('-');
+        if minus || s.starts_with('+') {
```
Comment (eddyb) [inline src/librustc_apfloat/ieee.rs]: This is the only change in this file that we should keep IMO (I guess I wasn't aware `starts_with` could take a `char`, when I did the port).

File: src/librustc_apfloat/ieee.rs (hunk 1 above)
Comment (JohnTitor) [reply src/librustc_apfloat/ieee.rs]: @eddyb Okay, I removed the change except for mentioned below.

**Review by eddyb — APPROVED** [review]

**Pull request merged**